prov/verbs/src/verbs_ep.c

```diff
@@ -36,6 +36,57 @@
 
 static struct fi_ops_msg fi_ibv_srq_msg_ops;
 
+
+ssize_t vrb_post_send(struct fi_ibv_ep *ep, struct ibv_send_wr *wr)
+{
+	struct vrb_context *ctx;
+	struct fi_ibv_cq *cq;
+	struct ibv_send_wr *bad_wr;
+	struct ibv_wc wc;
+	int ret, retry = 1;
+
+	cq = container_of(ep->util_ep.tx_cq, struct fi_ibv_cq, util_cq);
+	cq->util_cq.cq_fastlock_acquire(&cq->util_cq.cq_lock);
+	ctx = ofi_buf_alloc(cq->ctx_pool);
+	if (!ctx)
+		goto eagain;
+retry:
+	if (!cq->credits || !ep->tx_credits)
+		goto free_cq_space;
+
+	cq->credits--;
+	ep->tx_credits--;
+	cq->util_cq.cq_fastlock_release(&cq->util_cq.cq_lock);
+
+	ctx->ep = ep;
+	ctx->user_ctx = (void *) (uintptr_t) wr->wr_id;
+	wr->wr_id = (uintptr_t) ctx;
+
+	ret = ibv_post_send(ep->ibv_qp, wr, &bad_wr);
+	if (ret) {
+		cq->util_cq.cq_fastlock_acquire(&cq->util_cq.cq_lock);
+		cq->credits++;
+		goto free_cq_space;
```

Comment:
Looks like a missing close parentheses got dropped.

Reply:
How did this compile cleanly?  Odd...